src/box/space.c

```diff
@@ -576,8 +576,15 @@ space_create(struct space *space, struct engine *engine,
 			 "check_unique_constraint_map");
 		goto fail;
 	}
+
 	rlist_foreach_entry(index_def, key_list, link) {
-		struct index *index = space_create_index(space, index_def);
+		struct index_def *temp_def = index_def_dup(index_def);
+		free(temp_def->space_name);
+		temp_def->space_name = xstrdup(def->name);
+		strlcpy(temp_def->engine_name, def->engine_name,
+			ENGINE_NAME_MAX);		
+		struct index *index = space_create_index(space, temp_def);
+		index_def_delete(temp_def);
```

Comment:
This looks rather ugly :-/

Maybe, better add the two new arguments `space_name` and `engine_name` to `index_def_new()`? There aren't that many places where it's called so patching them all shouldn't be painful. The benefit of it is that we wouldn't have to think about possible `index_def::space_name` and `index_def::engine_name` changes - they'd stay the same the whole time until the object is deleted.

A side note: There's the `asan-ci` label you could use instead of `full-ci`. In contrast to `full-ci`, `asan-ci` only triggers ASAN workflows so CI should pass faster.